app/controllers/main/routes.py

```diff
@@ -118,6 +118,31 @@ def RemoveRSVP():
     flash("Successfully unregistered for event!", "success")
     return redirect(url_for("admin.editEvent", eventId=event.id))
 
+@main_bp.route('/<username>/serviceTranscript', methods = ['GET'])
+def serviceTranscript(username):
+    user = User.get_or_none(User.username == username)
+    if user is None:
+        abort(404)
+    if user != g.current_user and not g.current_user.isAdmin:
+        abort(403)
+
+    programs = getProgramTranscript(username)
+    slCourses, instructorDict = getSlCourseTranscript(username)
```

Comment:
We can simplify this method by either making the instructorDict use Course objects for keys, allowing you to loop through the one dictionary to get both courses and instructors, OR add an instructors backreference to CourseInstructor, which will let you access Course.instructors wherever you have a Course object